### PR TITLE
sql + storage: cleanup subsource dependency inversion code

### LINKED ISSUE #26764: sql+storage: remove legacy subsource planning/rendering elements
### Feature request

Once we release (not just merge) #26556, we should remove a number of now-vestigial structs from MZ. To find a first pass of them, grep this issue's number.

### PR DESCRIPTION
**Pull request by sploiselle**
Fixes #26764, #26769, #26772, #26768

### Motivation

 This PR refactors existing code.

### Tips for reviewer

The code touched is very disparate, probably go commit-by-commit and look for the crates that interest you.

### Checklist

- [ ] This PR has adequate test coverage / QA involvement has been duly considered. ([trigger-ci for additional test/nightly runs](https://trigger-ci.dev.materialize.com/))
- [ ] This PR has an associated up-to-date [design doc](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/README.md), is a design doc ([template](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/00000000_template.md)), or is sufficiently small to not require a design.
  
- [ ] If this PR evolves [an existing `$T ⇔ Proto$T` mapping](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/command-and-response-binary-encoding.md) (possibly in a backwards-incompatible way), then it is tagged with a `T-proto` label.
- [ ] If this PR will require changes to cloud orchestration or tests, there is a companion cloud PR to account for those changes that is tagged with the release-blocker label ([example](https://github.com/MaterializeInc/cloud/pull/5021)).
  
- [ ] This PR includes the following [user-facing behavior changes](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/guide-changes.md#what-changes-require-a-release-note):
  -

### CODE REVIEW

File: src/sql/src/pure/mysql.rs [hunk 1]
```diff
@@ -193,11 +194,15 @@ pub(super) fn normalize_column_refs<'a>(
...
         let (column_name, qual) = name.0.split_last().expect("non-empty");
         match catalog.resolve(UnresolvedItemName::qualified(qual)) {
+            // TODO: this needs to also introduce the maximum qualification on
+            // the columns, i.e. ensure they have the schema name.
             Ok((_, desc)) => desc.columns.iter().any(|n| &n.name == column_name.as_str()),
             Err(_) => false,
         }
     });
 
+    tracing::info!("normalize_column_refs {:?}", seq);
```
Comment (rjobanp) [inline src/sql/src/pure/mysql.rs:204]: ditto

File: src/sql/src/pure.rs [hunk 2]
```diff
@@ -124,6 +120,12 @@ fn subsource_gen<'a, T>(
 
         let (qualified_upstream_name, desc) = catalog.resolve(subsource.reference.clone())?;
 
+        tracing::info!(
```
Comment (rjobanp) [inline src/sql/src/pure.rs:123]: should this be info?

File: src/sql/src/pure.rs [hunk 3]
```diff
@@ -1125,6 +1119,8 @@ async fn purify_create_source(
             let new_subsources =
                 mysql::generate_targeted_subsources(&scx, validated_requested_subsources)?;
 
+            tracing::info!("new_subsources {:?}", new_subsources);
```
Comment (rjobanp) [inline src/sql/src/pure.rs:1122]: ditto

File: src/sql-parser/src/parser.rs [hunk 4]
```diff
@@ -2622,13 +2622,12 @@ impl<'a> Parser<'a> {
     fn parse_create_subsource_option_name(
```
Comment (rjobanp) [inline src/sql-parser/src/parser.rs:2622]: the docstring seems incorrect for this method

**Review by rjobanp — APPROVED** [review]
nice cleanup! this LGTM from a storage perspective besides the minor things I noted

File: src/sql/src/pure/mysql.rs [hunk 5]
```diff
@@ -88,6 +84,11 @@ pub(super) fn generate_targeted_subsources(
         table,
     } in validated_requested_subsources.into_iter()
     {
+        tracing::info!(
```
[the remark this answers is not in the document]
Comment (sploiselle) [reply src/sql/src/pure/mysql.rs:87]: Oh gosh--yes. This is what I get for pushing code after 10pm. Will remove.

File: src/sql-parser/src/parser.rs (hunk 4 above)
Comment (sploiselle) [reply src/sql-parser/src/parser.rs:2622]: Fixed!

**Pull request merged**